clj/src/bucket_distributor_clj/core.clj

```diff
@@ -0,0 +1,468 @@
+(ns bucket-distributor-clj.core
+  (:require [clojure.tools.logging :as log]
+            [clojure.core.async :as async :refer [chan <!! >!!]]
+            [langohr.core      :as rmq]
+            [langohr.channel   :as lch]
+            [langohr.queue     :as lq]
+            [langohr.exchange  :as le]
+            [langohr.consumers :as lc]
+            [langohr.basic     :as lb]
+            [midje.sweet       :as t])
+  (:import [java.util.concurrent LinkedBlockingQueue TimeUnit Executors ScheduledExecutorService]))
+
+(defn- ex 
+  [msg] (Exception. msg))
+
+(defn require-ch 
+  [conn]
+  (let [ch (lch/open conn)]
+    (when-not ch
+      (throw (Exception. "cannot open channel")))
+    ch))
+
+(defmacro with-conn
```

Comment:
I made these macros because langhor doesn't support Closeable, and also because the close semantics on connections and channels aren't normal. (Certain operations can forcibly close a channel if they fail, and once this happens calling close() then fails. Its a weird api.)